mock/src/main.rs

```diff
@@ -0,0 +1,105 @@
+use std::{fs::File, io::Write, path::PathBuf, time::Instant};
+
+use clap::Parser;
+use crypto::hash::rpo::RpoDigest as Digest;
+use miden_objects::{
+    builder::DEFAULT_ACCOUNT_CODE,
+    mock::{Immutable, MockChain, OnChain},
+};
+use rand::SeedableRng;
+use rand_pcg::Pcg64;
+
+#[derive(Parser, Debug)]
+struct Args {
+    /// Value used to seed the PRNG.
+    #[arg(long)]
+    seed: Option<u64>,
+
+    /// Output file
+    output: PathBuf,
+}
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    env_logger::init();
+
+    let args = Args::parse();
+
+    let small_rng = match args.seed {
+        Some(seed) => Pcg64::seed_from_u64(seed),
+        None => Pcg64::from_entropy(),
+    };
+
+    println!(
+        "Generating mock data, results will be saved to {}",
+        args.output.to_str().expect("invalid file name")
+    );
+
+    let mut output = File::create(args.output)?;
+
+    // Block 1
+    // --------------------------------------------------------------------------------------------
+    let mut mock_chain = MockChain::new(small_rng);
+    let start = Instant::now();
+    let faucet =
+        mock_chain.build_fungible_faucet(OnChain::Yes, DEFAULT_ACCOUNT_CODE, Digest::default());
+    println!("Fungible faucet created {} [took: {}s]", faucet, start.elapsed().as_secs());
+    mock_chain.seal_block();
+
+    // Block 2
+    // --------------------------------------------------------------------------------------------
+    let amount = 100;
+    let asset = mock_chain.build_fungible_asset(0, amount);
+
+    let storage = vec![];
+    let assets = vec![asset];
```

Comment:
One of the validation criteria for new account creation is that the vault is empty.  As such this is not suitable.

https://github.com/0xPolygonMiden/miden-base/blob/5db74919399336df2283e3e12a45988897bb76e7/miden-lib/asm/sat/internal/prologue.masm#L157-L168

Reply:
Thanks for the heads up! Why is that restriction in place? Bootstrapping the genesis block is a bit harder since we can't just aidrop tokens to the account (as it is done here). It would be useful to have that.

I suppose the alternative is to use the faucet's storage to map the id to its airdropped amount, and then the user would need to do a transaction to retrieve it, that does cause issues with bootstrapping users though (one would need to have tokens to get tokens).

Reply:
It's to ensure that all minting of assets passes through the minting logic we have in the kernel. 

Yes I think your proposal of using the storage (maybe a child smt) to store the airdrop amounts is a viable solution but this does require an additional step and quite a few additional clock cycles.

Reply:
Is the cold start problem related to needing to have funds to pay fees? If so, we don't have fees for now - so, should be OK. In the future, we could have faucets pay fees for user transactions.

For example, the distribution logic on a faucet could include logic for making an appropriate fee payment for a qualified account. As long as these are network transactions, this should work fine, I think.

Reply:
@bobbinth not really, AFAIU there are no requirements for the time being for gas payments.

It was just simpler writing it this way, and that is what I understood you requested.